core/src/com/unciv/logic/map/tile/TileImprovementFunctions.kt

```diff
@@ -311,31 +311,32 @@ class TileImprovementFunctions(val tile: Tile) {
         val closestCity = civ.cities.minByOrNull { it.getCenterTile().aerialDistanceTo(tile) }
             ?: return
         val distance = closestCity.getCenterTile().aerialDistanceTo(tile)
-        var productionPoints = 0f
+        var stats = Stats()
         for (unique in tile.getTerrainMatchingUniques(UniqueType.ProductionBonusWhenRemoved)) {
-            productionPoints = unique.params[0].toFloatOrNull() ?: return
+            //productionPoints = unique.params[0].toFloatOrNull() ?: return
+            stats.add(unique.stats)
         }
```

Comment:
If stats is empty, you should quit here. Otherwise you'll get empty notifications every time you clear any feature

Reply:
The return should be after the loop, maybe one of the other uniques has something :)

Reply:
You mean it doesn't go straight to the respective unique, but iterates over all the uniques?

Reply:
Yes, if it has several uniques it should give them all
E.g. bonus chop stats with conditionals